.github/workflows/copilot_deploy.yml

```diff
@@ -50,63 +50,69 @@ jobs:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        value: ${{ fromJSON(needs.copilot_environments_workflow_setup.outputs.matrix) }}
+    environment: ${{ matrix.value }}
+    steps:
+    - name: Git clone the repository
+      uses: actions/checkout@v4
 
-  # Allow the capability to override UAT with another branch, but ideally uat and production should be in sync as much as possible
-  uat_copilot_deploy:
-    if: inputs.environment == 'uat' || inputs.environment == ''
-    needs: [pre_deploy_tests, paketo_build]
-    concurrency: deploy-uat
-    secrets:
-      AWS_ACCOUNT: ${{ secrets.AWS_ACCOUNT }}
-    uses: ./.github/workflows/environment.yml
-    permissions:
-      id-token: write # This is required for requesting the JWT
-      contents: read  # This is required for actions/checkout
-    with:
-      workspace: 'uat'
+    - name: Get current date
+      shell: bash
+      id: currentdatetime
+      run: echo "datetime=$(date +'%Y%m%d%H%M%S')" >> $GITHUB_OUTPUT
 
-  # Only run this if the branch being deployed is main
-  production_copilot_deploy:
-    if: (inputs.environment == 'production' || inputs.environment == '') && github.ref == 'refs/heads/main'
-    needs: [pre_deploy_tests, paketo_build]
-    concurrency: deploy-production
-    secrets:
-      AWS_ACCOUNT: ${{ secrets.AWS_ACCOUNT }}
-    uses: ./.github/workflows/environment.yml
-    permissions:
-      id-token: write # This is required for requesting the JWT
-      contents: read  # This is required for actions/checkout
-    with:
-      workspace: 'production'
+    - name: configure aws credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT }}:role/GithubCopilotDeploy
+        role-session-name: NOTIFICATION_${{ matrix.value }}_COPILOT_${{ steps.currentdatetime.outputs.datetime }}
+        aws-region: eu-west-2
+
+    - name: Install AWS Copilot CLI
+      shell: bash
+      run: |
+        curl -Lo aws-copilot https://github.com/aws/copilot-cli/releases/latest/download/copilot-linux && chmod +x aws-copilot && sudo mv aws-copilot /usr/local/bin/copilot
+
+    - name: Inject Git SHA into manifest
+      run: |
+        yq -i '.variables.GITHUB_SHA = "${{ github.sha }}"'  copilot/fsd-fund-store/manifest.yml
+
+    - name: Inject replacement image into manifest
+      run: |
+        yq -i '.image.location = "ghcr.io/communitiesuk/funding-service-design-fund-store:${{ github.ref_name == 'main' && 'latest' || github.ref_name }}"'  copilot/fsd-fund-store/manifest.yml
+
+    - name: Copilot ${{ matrix.value }} deploy
+      id: deploy_build
+      run: |
+        copilot svc deploy --env ${{ matrix.value }} --app pre-award
 
   # Can we realistically run E2E at this stage, or just plump for application on the grounds it checks fund-store is operational?
   post_deploy_tests:
-      needs: test_copilot_deploy
+      needs: copilot_environments_deploy
```

Comment:
Post Deploy Tests should only run if test has changed - which would be if the test environment is targeted (manually) or it's going through the automated functionality